src/logtracing/Logger.py

```diff
@@ -1,20 +1,107 @@
 import os
+import sys
+import json
+import time
+import psutil
+import platform
 import traceback
 from typing import Union
-from Types import PrepareStackTrace, CodeLine, ErrorStack, OsVars, PythonVars
+from Types import CodeLine, ErrorStack, PythonVars, OsVars, PrepareStackTrace
 
 
 class Logger:
     def __init__(self, flow: str):
-      self._flow = flow;
-      self.prepare_stack_trace: PrepareStackTrace = traceback.extract_stack
-      self.err_stack = None;
-      self.os_vars: Union[OsVars, None] = None;
-      self.python_vars: Union[PythonVars, None] = None;
-      self.env_vars: Union[os.environ, None] = None;
-      self.extra_vars = {};
-      self.code_lines_limit: int = 5;
+        if not flow:
+            raise Exception('Flow argument is missing')
+
+        self._flow = flow
+        self.prepare_stack_trace: PrepareStackTrace = traceback.extract_stack
+        self.err_stack: Union[ErrorStack, None] = None
+        self.os_vars: Union[OsVars, None] = None
+        self.python_vars: Union[PythonVars, None] = None
+        self.env_vars: Union[os.environ, None] = None
+        self.extra_vars = {}
+        self.code_lines_limit: int = 5
 
     @property
     def flow(self):
         return self._flow
+
+    def track_error(self, err) -> None:
+        if not err:
+            raise Exception('Error argument is missing')
+
+    def report(self) -> None:
+        self.load_os_vars()
+        self.load_python_vars()
+        self.load_env_vars()
+
+        print(self.err_stack)
+        print(self.os_vars)
+        print(self.env_vars)
+        print(self.python_vars)
+        print(self.extra_vars)
+
+    def add_extra(self, identifier: str, extra) -> None:
+        if not isinstance(extra, dict) and not isinstance(extra, str):
```

Comment:
It could be more compact as:
```suggestion
        if not isinstance(extra, (dict, str)):
```

Reply:
I changed it, thank you!